adaboost/tests/test_cuda_operations.cu

```diff
@@ -0,0 +1,44 @@
+#include<gtest/gtest.h>
+#include<string>
+#include<adaboost/cuda/core/cuda_data_structures.hpp>
+#include<adaboost/cuda/utils/cuda_wrappers.hpp>
+#include<adaboost/core/operations.hpp>
+#include<adaboost/cuda/core/operations.hpp>
+#include<stdexcept> 
+#include <cuda.h>
+
+
+__device__ float square_1(float x)
+{
+    return x*x;
+}
+
+__device__ float square_2(float x)
+{
+    return -x*x;
+}
+
+__device__ const func_t<float,float> p_func=square_1;
+
+TEST(Cuda, Argmax)
+{
+    adaboost::utils::cuda::cuda_event_t has_happened;
+    adaboost::utils::cuda::cuda_event_create(&has_happened);
+    adaboost::cuda::core::VectorGPU<float> vec_i(5);
+    vec_i.set(0, -1.0);
+    vec_i.set(1, 0);
+    vec_i.set(2, 1);
+    vec_i.set(3, 2);
+    vec_i.set(4, 3);
+    unsigned int block_size = 2;
+    adaboost::utils::cuda::cuda_event_record(has_happened);
+    adaboost::utils::cuda::cuda_event_synchronize(has_happened);
+    float result_gpu;
+    adaboost::cuda::core::Argmax(p_func, vec_i, result_gpu, 2);
```

Comment:
Is this test working locally?

Reply:
If I pass 0, ie use Argmax from adaboost::core, it gives the correct answer.

Reply:
Well, we are interested in `block_size > 0`. :-)
In any case, `block_size<=0` shouldn't be allowed in CUDA APIs. I did allow them when I started this project but since after changing the code structure, CUDA APIs are different from CPU APIs, so if someone wants to use CPU ones they can directly use it rather than passing `block_size=0`. For now you can just throw a logical error in `Argmax` in case `block_size<=0` and some similar errors can be thrown in other operations under `adaboost::cuda` after this PR is merged.

Reply:
Ok, will make the change